File: hw/1/khlybova/app/src/main/java/com/nsu/learning/myapplication/view/MenuFragment.java
package com.nsu.learning.myapplication.view;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.nsu.learning.myapplication.R;


public class MenuFragment extends AppFragment {
    public MenuFragment() {
        // Required empty public constructor
    }


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_menu, container, false);
        setStartFragmentButton(view.findViewById(R.id.button_start_game), new ChooseGamersFragment());
        setStartFragmentButton(view.findViewById(R.id.button_score), new ScoreFragment());
        return view;
    }

}

File: hw/1/TicTacToe/app/src/main/java/ru/brz/tictactoe/MenuFragment.kt
package ru.brz.tictactoe

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.activityViewModels
import androidx.navigation.Navigation
import ru.brz.tictactoe.databinding.FragmentMenuBinding

class MenuFragment : Fragment() {

    private lateinit var binding: FragmentMenuBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentMenuBinding.inflate(inflater)

        val model: GameViewModel by activityViewModels()
        if(model.player1.value == null){
            binding.buttonResume.visibility = View.GONE
        }

        binding.buttonNewGame.setOnClickListener {
            Navigation.findNavController(it).navigate(MenuFragmentDirections.actionMenuFragmentToLobbyFragment())
        }
        binding.buttonResume.setOnClickListener {
            Navigation.findNavController(it).navigate(MenuFragmentDirections.actionMenuFragmentToGameFragment())
        }
        binding.buttonScore.setOnClickListener {
            Navigation.findNavController(it).navigate(MenuFragmentDirections.actionMenuFragmentToScoresFragment())
        }

        return binding.root
    }

}